buch/src/components/LoginBox/index.tsx

```diff
@@ -4,15 +4,23 @@ import "./index.scss";
 
 interface IProps {
     handleConfirm: (username: string, password: string) => void;
+    handleRegister: (username: string, password: string) => void;
     handleCancel: () => void;
 }
 
 export default function LoginBox(props: IProps) {
     const [username, setUsername] = useState("");
     const [password, setPassword] = useState("");
+
+    const [showErrorMessage, setShowErrorMessage] = useState(false);
+
     const confirm = () => {
+        username == "" && setShowErrorMessage(true);
```

Comment:
function just doesn't cancel if there is no username. Wouldn't even show an error message if there was one, because it instantly closes the LoginBox